.github/workflows/pr-check.yml

```
@@ -13,4 +13,6 @@ jobs:
         run: |
           npm ci
           npm run lint
+          npm run build
+          cp wrangler.toml.template wrangler.toml
           npm run test
```

@Aprillion does the CI have access to a `CODA_TOKEN`? If so, perhaps it would be good to refresh the test data during CI, to make sure that tests are running against the latest data from Coda. This might make the tests more flaky and unpredictable, but it would be a better test that the site will work against current data 🤔 .
```suggestion
          npm run build
          cp wrangler.toml.template wrangler.toml
          npm run refresh-test-data
          npm run test
```

---

yes, GH actions have access to `${{ secrets.CODA_TOKEN }}` like in https://github.com/StampyAI/stampy-ui/blob/master/.github/workflows/deploy.yml

---

Ok, thanks. I propose that we run both cases during the CI workflow:
1. Run the tests against that the cached, git-committed data.
2. Refresh the data using the CODA_TOKEN and then run the tests (without committing the refresh)

In this way:
- if number 1 fails, this would suggest a regression bug in some code changes
- if number 1 passes but number 2 fails, this suggests perhaps flakiness in the coda query or that the code needs to be changed to be updated to match the latest data
- if, after a code change, number 1 fails, but number 2 passes, this could suggest that the cached data needs updated

I think that while developing, using cached data for the tests would help them run faster.

What are your thoughts on the above?

---

hm, I like having 2 commands, so that locally I will update only once and then run the cached data

but not sure that I see the benefit of running on old data in CI, I'd be happy to only run it after refresh..

---

Ok, great. I changed
- the pr-check workflow to export CODA_TOKEN as an env variable and `refresh-coda-data-for-tests` to take it from the env or from the wrangler config.
- the pr-check workflow also now runs `npm run refresh-cached-data` so that the tests are run on the latest data